tests/openshell/conftest.py

```
@@ -0,0 +1,318 @@
+    """
+    provider_name = OPENSHELL_VLLM_PROVIDER
+
+    provider = datamodel_pb2.Provider(
+        metadata=datamodel_pb2.ObjectMeta(name=provider_name),
+        type="openai",
+        config={"OPENAI_BASE_URL": OPENSHELL_VLLM_ENDPOINT},
+        credentials={"OPENAI_API_KEY": OPENSHELL_VLLM_TOKEN},
+    )
+
+    LOGGER.info("Creating vLLM inference provider", name=provider_name, endpoint=OPENSHELL_VLLM_ENDPOINT)
+    sandbox_client._stub.CreateProvider(  # noqa: FCN001
```

Ruff (0.16.1)

[warning] Invalid rule code in `# noqa`: FCN001

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)